test/lib/visualTestLoop.js

```diff
@@ -106,6 +106,31 @@
     img.src = filename;
   }
 
+  function dumpFailedTest(testName, original, golden, difference) {
+    if (fabric.isLikelyNode && original && difference && golden) {
+      var largeCanvas = fabric.util.createCanvasElement();
+      largeCanvas.width = original.width + golden.width + difference.width;
+      largeCanvas.height = Math.max(original.height, golden.height, difference.height);
+      var ctx = largeCanvas.getContext('2d');
+      ctx.drawImage(original, 0, 0);
+      ctx.putImageData(difference, original.width, 0);
+      ctx.drawImage(golden, original.width + difference.width, 0);
+      var dataUrl = largeCanvas.toDataURL().split(',')[1];
+      console.log('dumping failed test', testName);
+      const fileName = localPath('../../cli_output', `${testName.replaceAll(' ', '_')}.png`);
+      fs.writeFileSync(fileName.replace('file://', ''), dataUrl, { encoding: 'base64' });
+    }
+    // else if (original) {
```

Comment:
leave this please
I use it, git shows a very convenient diff.
And this way I just commit if I need to

![image](https://user-images.githubusercontent.com/34343793/187117718-4ec9f0e0-59d1-4b81-8129-36f7511f5535.png)


![image](https://user-images.githubusercontent.com/34343793/187117670-24521dde-a858-45b6-8815-ed6b52aa9108.png)

Reply:
that is not the one you use. That is a copy that i commented because i didn't write for browser

Reply:
https://github.com/fabricjs/fabric.js/pull/8206/files#diff-e236a8dadb89494460b0e0c2d0018e76d602cb4bf08e1f52d3e673b7709a03ebR71-R79

this is the one you mean

Reply:
> https://github.com/fabricjs/fabric.js/pull/8206/files#diff-e236a8dadb89494460b0e0c2d0018e76d602cb4bf08e1f52d3e673b7709a03ebR71-R79
> 


The link pints me to a strange place. Add a comment in the code so I can see


> that is not the one you use. That is a copy that i commented because i didn't write for browser

I wrote that in #8138

Reply:
```
I have disabled golden debugging/recreation from node because I understood that it is not accurate enough because of node-canvas
```
Yes node-canvas is not accurate enough, but i generate the golden from node anyway because they help finding out issues earlier. When a test is too much different in node, we disable the test for node ( we need to do that anyway or it will fail ).

```
Don't you enjoy using it this way?
```

I think the old was the best, i deleted a golder and i would get it back just running the tests, no flags.

Reply:
> I think the old was the best, i deleted a golden and i would get it back just running the tests, no flags.

It still works like that.
The `-d` flag will recreate the golden **ONLY** if the test failed
The `-r` flag will recreate the golden regardless

Reply:
I don't mind switching the `-d` flag on as default

Reply:
For me it is annoying running a test, seeing it fails, navigating to the file, deleting it and all over again each time I change something in the code and want to test the diff

I flag `-d` and it happens.
Then I can discard changes in github desktop.
No hassle.

Reply:
Perhaps I didn't explain how to use it properly until now.
It should make your life much easier.